File: sdk/typescript/src/catalog-io.ts
/** Catalog disk I/O and builder (Rust-backed). */

import { CatalogIndex } from "./build.js";
import {
  CatalogBuilderNative,
  loadCatalogIndexFromDirNative,
  writeCatalogIndexNative,
} from "./native.js";
import type { JsonRecord } from "./types.js";

export function writeCatalogIndex(
  index: JsonRecord,
  outputDir?: string | null,
  prune?: boolean | null,
): void {
  writeCatalogIndexNative(index, outputDir ?? undefined, prune ?? undefined);
}

export function loadCatalogIndexFromDir(dirPath: string): CatalogIndex {
  const result = loadCatalogIndexFromDirNative(dirPath) as {
    tools: JsonRecord[];
    files: Record<string, string>;
  };
  return new CatalogIndex(result.tools, result.files);
}

export type CatalogBuilder = InstanceType<typeof CatalogBuilderNative>;

export { CatalogBuilderNative as CatalogBuilder };

File: sdk/go/catalog_io.go
package chunkyourtools

// WriteCatalogIndex writes a catalog index JSON snapshot to a directory.
func WriteCatalogIndex(indexJSON, outputDir string, prune bool) error {
	return cgoWriteCatalogIndex(indexJSON, outputDir, prune)
}

// LoadCatalogIndexFromDir reads a catalog index from a directory on disk.
func LoadCatalogIndexFromDir(dirPath string) (string, error) {
	return cgoLoadCatalogIndexFromDir(dirPath)
}
